### PR TITLE
OnBoarding: new features

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Motivation

We need:
* More logs related with the installation and weblog
* Avoid creating new provision, only to add one env parameter
* Avoid duplicated code to create app services 

## Changes

* All weblog's stdout to /var/log/datadog_weblog
* Add log extractor step: /var/log/datadog and /var/log/datadog_weblog
* Common script to create weblog service
* Env variables for weblog or agent installation script set in the scenario definition (allow us simple way to activate features like ASM or PROFILING) 

This PR pretended to be more than one PR, but finally I added all new features in one PR, because all are related.
No applied to container scenarios. This will come in separate PR

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml [hunk 1]
```diff
@@ -7,6 +7,14 @@
       local_path: utils/build/virtual_machine/provisions/auto-inject/docker/docker_config.yaml
 
   remote-command: |
+
+    #Env variables set on the scenario definition. Write to file and load  
+    SCENARIO_AGENT_ENV="${DD_AGENT_ENV:-''}"
+    echo "${SCENARIO_AGENT_ENV}" > scenario_agent.env
+    echo "AGENT VARIABLES CONFIGURED FROM THE SCENARIO:"
+    cat scenario_agent.env
+    export $(cat scenario_agent.env | xargs)
```
Comment (BaptisteFoy) [inline utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:16]: Isn't it easier to use something like `eval`?

**Review by BaptisteFoy — APPROVED** [review]
LGTM if tests pass! Thanks for all the new features

**Review by szegedi — APPROVED** [review]
Thank you for the env vars configuration feature! It was indeed fragile to maintain copies of very similar scripts. I'm glad to see that you not only managed to eliminate my `-profiling` variants, but was able to eliminate language-specific instances of `test-app.service` files and replace them with a single version.

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml (hunk 1 above)
Comment (robertomonteromiguel) [reply utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:16]: You are probably right. At this point I was stuck for several hours.

**Pull request merged**